pysb/tools/sensitivity_analysis.py

```diff
@@ -0,0 +1,603 @@
+                    bp_not_in_b_dict[counter] = bp_not_in_b_visited[s_1]
+                else:
+                    new_sim_counter += 1
+                    x = self._index_of_species_of_interest[index_i]
+                    y = self._index_of_species_of_interest[index_j]
+                    b_prime[new_sim_counter, x] *= sigma_i
+                    b_prime[new_sim_counter, y] *= sigma_j
+                    bp_not_in_b_visited[s_1] = new_sim_counter + n_b_index
+                    bp_not_in_b_dict[counter] = new_sim_counter + n_b_index
+                    bp_not_in_b_raw.add(new_sim_counter)
+
+        self.b_prime_in_b = bp_dict
+        self.b_prime_not_in_b = bp_not_in_b_dict
+        x = b_to_run[list(self.b_index)]
+        y = b_prime[list(bp_not_in_b_raw)]
+        simulations = np.vstack((x, y))
+        print("Number of simulations to run = %s" % len(simulations))
+        return simulations
+
+    def create_plot_p_h_pprime(self, save_name=None, out_dir=None, show=False):
+        """
+        Plot of P, H(B), and P'
+
+        See :class:`InitialsSensitivity` for descriptions of these matrices
+
+        Parameters
+        ----------
+        save_name : str, optional
+            name to save figure as
+        out_dir : str, optional
+            location to save figure
+        show : bool
+            show the plot if True
+        """
+        colors = 'seismic'
+        sens_matrix = self.p_matrix - self.p_prime_matrix
+        v_max = max(np.abs(self.p_matrix.min()), self.p_matrix.max())
+        v_min = -1 * v_max
+
+        fig = plt.figure(figsize=(12, 6))
+        ax1 = fig.add_subplot(131)
+        ax2 = fig.add_subplot(132)
+        ax3 = fig.add_subplot(133)
+        ax1.imshow(self.p_matrix, interpolation='nearest', origin='upper',
+                   cmap=plt.get_cmap(colors), vmin=v_min, vmax=v_max,
+                   extent=[0, self._nm, 0, self._nm])
+
+        ax2.imshow(self.p_prime_matrix, interpolation='nearest',
+                   origin='upper', cmap=plt.get_cmap(colors),
+                   vmin=v_min, vmax=v_max, extent=[0, self._nm, 0, self._nm])
+
+        ax3.imshow(sens_matrix, interpolation='nearest', origin='upper',
+                   cmap=plt.get_cmap(colors), vmin=v_min, vmax=v_max)
+
+        for ax in (ax1, ax2, ax3):
+            ax.set_xticks([])
+            ax.set_yticks([])
+
+        ax1.set_title('P', fontsize=22)
+        ax2.set_title('H(B\')', fontsize=22)
+        ax3.set_title('P\' = P - H(B\')', fontsize=22)
+        fig.subplots_adjust(wspace=0, hspace=0.0)
+        if save_name is not None:
+            if out_dir is None:
+                out_dir = '.'
+            if not os.path.exists(out_dir):
+                os.mkdir(out_dir)
+            fig.savefig(os.path.join(out_dir,
+                                     '{}_P_H_P_prime.png'.format(save_name)),
+                        bbox_inches='tight')
+        if show:
+            plt.show()
+        plt.close()
```

Comment:
Could return `fig` here just in case someone wants to do something with it you didn't anticipate. Same goes for the other plotting functions below.